internal/proxy/accesslog/minio_handler.go

```diff
@@ -0,0 +1,229 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package accesslog
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/milvus-io/milvus/internal/log"
+	"github.com/milvus-io/milvus/internal/util/retry"
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"go.uber.org/zap"
+)
+
+type config struct {
+	address           string
+	bucketName        string
+	accessKeyID       string
+	secretAccessKeyID string
+	useSSL            bool
+	createBucket      bool
+	rootPath          string
+	useIAM            bool
+	iamEndpoint       string
+}
+
+//minIO client for upload access log
+//TODO file retention on minio
```

Comment:
is the `TODO` finished?

Reply:
No, this pr was too large, and I want to do this at next one.

Reply:
> is the `TODO` finished?

No, this pr was too large, and I want to do this at next one.